src/components/form/fields/identifierField.jsx

```diff
@@ -78,7 +78,7 @@ class IdentifierField extends React.Component {
             country,
             formatted,
             isValid,
-            raw: phone,
+            raw: isValid ? phone : inputValue,
```

Comment:
For input `1234abc`, `phone` would be `1234`. If the phone is not valid, use the raw string as a value. That's what was keeping characters other than numbers from being entered. They were taken into account but the formatted version discarded them, and this was used to render the field.